presto-spi/src/main/java/com/facebook/presto/spi/block/ByteArrayBlockBuilder.java

```diff
@@ -247,4 +262,10 @@ private void checkReadablePosition(int position)
             throw new IllegalArgumentException("position is not valid");
         }
     }
+
+    // Should the above block be replaced with a RLEBlock representing null values.
+    private boolean canBeReplacedWithRLEBlock()
```

Comment:
There is no need for an extra method for this.  `if (!hasNonNullValue)` is very readable